modin/core/dataframe/algebra/groupby.py

```diff
@@ -289,31 +289,37 @@ def caller(
         The same type as `query_compiler`
             QueryCompiler which carries the result of GroupBy aggregation.
         """
+        is_unsupported_axis = axis != 0
+        is_empty_data = (
+            len(query_compiler.columns) == 0 or len(query_compiler.index) == 0
```

Comment:
Will it trigger calculations? Or will they be triggered somewhere before that?

Reply:
Well, it will trigger computations in case columns or indices are deferred. But we have to determine whether the data is empty here as we can't process empties anyhow. So, here we're potentially sacrificing laziness in terms of stability.

The other thing is that currently there is no way we may end up at the query compiler's groupby implementation without an index being triggered to compute (it will be triggered anyway at [dataframe.groupby()](https://github.com/modin-project/modin/blob/b8b58cc45827ceb32f9f493622d5e66d37ab7502/modin/pandas/dataframe.py#L436) and probably in other places). So I propose to think about groupby laziness when it would become possible for real.

Reply:
Got it. Ok.